tests/test_build.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+import os
+import shutil
+
+import pytest
+
+from piptools.build import build_project_metadata
+from tests.constants import PACKAGES_PATH
+
+
+@pytest.mark.network
+def test_build_project_metadata_resolved_correct_build_dependencies(
+    fake_dists_with_build_deps, tmp_path, monkeypatch
+):
+    """Test that the resolved build dependencies are correct.
+
+    Because this is a slow process we test it only for one build distribution and rely
+    on ``test_all_extras_and_all_build_deps`` to test that it works with multiple build
+    distributions.
+    """
+    # When used as argument to the runner it is not passed to pip
+    monkeypatch.setenv("PIP_FIND_LINKS", fake_dists_with_build_deps)
+    src_pkg_path = os.path.join(PACKAGES_PATH, "small_fake_with_build_deps")
+    shutil.copytree(src_pkg_path, tmp_path, dirs_exist_ok=True)
+    src_file = tmp_path / "setup.py"
+    metadata = build_project_metadata(
+        os.fspath(src_file), ("editable",), isolated=True, quiet=False
+    )
+    assert sorted(r.name for r in metadata.build_requirements) == [
```

Comment:
```suggestion
    build_requirements = sorted(r.name for r in metadata.build_requirements)
    assert build_requirements == [
```

Interim variables might be useful when debugging code.

Reply:
Will fix